### PR TITLE
Add ping pong and pingtimeout for ws connection

### PR DESCRIPTION
**Pull request by lukasIO**

### CODE REVIEW

**Review by boks1971 — APPROVED** [review]
Let's confirm with @davidzhao about direction and impact for other clients.

File: livekit_rtc.proto [hunk 1]
```diff
@@ -33,6 +33,8 @@ message SignalRequest {
     SyncState sync_state = 12;
     // Simulate conditions, for client validations
     SimulateScenario simulate = 13;
+    // client triggered ping to server
+    Ping ping = 14;
```
Comment (davidzhao) [inline livekit_rtc.proto:37]: what about sending an int64 instead?
```suggestion
    int64 ping = 14;
```

that way when pong is received, we could compute delta on time spent.

File: livekit_rtc.proto [hunk 2]
```diff
@@ -71,6 +73,8 @@ message SignalResponse {
     string refresh_token = 16;
     // server initiated track unpublish
     TrackUnpublishedResponse track_unpublished = 17;
+    // respond to ping
+    bool pong = 18;
```
Comment (davidzhao) [inline livekit_rtc.proto:77]: same as above

**Comment by davidzhao** [comment]
Updated JS client generation to use union types so it's no longer required to wrap within another type.

**Pull request merged**